src/FRP/Event/Time.purs

```diff
@@ -15,3 +17,22 @@ foreign import animationFrame :: Event Unit
 
 -- | Create an event which reports the current time in milliseconds since the epoch.
 foreign import withTime :: forall a. Event a -> Event { value :: a, time :: Int }
+
+-- | Similar to `sampleOn_`, except that the returned `Event` is a `Tuple` of
+-- | the sampled value _as well as_ the time since that value was pushed to the
+-- | stream (in milliseconds).
+withDelay :: forall a b. Event a -> Event b -> Event (Tuple Int a)
```

Comment:
In my case:

- You have an animation that lasts for one second
- Sample the event `withDelay` to get the original firing time
- `delay / 1000` = progress through animation, `> 1` = finished